tools/python/boutiques/creator.py

```diff
@@ -34,14 +40,64 @@ def __init__(self, parser=None, **kwargs):
             del self.descriptor["error-codes"]
             if type(parser) is not argparse.ArgumentParser:
                 raise CreatorError("Invalid argument parser")
-
             self.parseParser(**kwargs)
 
     def save(self, filename):
         import json
         with open(filename, "w") as f:
             f.write(json.dumps(self.descriptor, indent=4, sort_keys=True))
 
+    def parse_docker(self, descriptor, image_name, use_singularity):
+        cont_image = {}
+
+        # Basic image config
+        if use_singularity:
+            cont_image['type'] = 'singularity'
+            cont_image['index'] = 'docker://'
+            cont_image['image'] = image_name
+        else:
+            cont_image['type'] = 'docker'
+            cont_image['image'] = image_name
+
+        # Properties found in the image metadata
+
+        ((stdout, stderr),
```

Comment:
Look into using singularity instead of Docker, if the user specified a singularity image.

Reply:
http://singularity.lbl.gov/docs-environment-metadata

Reply:
Good idea! 
Note to self: here is how to find the entry point of a Singularity container:
```
$ singularity exec ./mysql.simg cat /.singularity.d/runscript
#!/bin/sh

exec "docker-entrypoint.sh" "$@"
```

Reply:
In a first step, let's try to find metadata using Docker, and exit gracefully if Docker is not installed (or fall back on Singularity).